### PR TITLE
[BD-46] docs: improve documentation for components

### PR DESCRIPTION
**Pull request by monteri**
improve documentation for MailToLink, Marketing Modal, Menu, ModalDialog, ModalLayer, ModalPopup, Navbar components

JIRA: [PAR-578](https://openedx.atlassian.net/browse/PAR-578)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @monteri! I've created [BLENDED-1018](https://openedx.atlassian.net/browse/BLENDED-1018) to keep track of it in Jira.   More details are on the [BD-46](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3134685276/BD-46+Paragon+Blended+Support+and+Maintenance) project page.


When this pull request is ready, tag your edX technical lead.

File: src/Modal/marketing-modal.mdx [hunk 1]
```diff
@@ -13,7 +13,15 @@ notes: |
 
 ---
 
-The composition of `ModalDialog` for marketing style scenarios. This component passes through all of its props to an underlying `ModalDialog` component providing some limited customization. If you have unique needs, use the `ModalDialog` compound component family directly.
+Marketing modal is similar to the alert modal provides friendly messaging to our users to encourage greater engagement than the simpler confirmation dialogs. They should enable platform understanding, highlight opportunities and support workflow engagement.
```
Comment (adamstankiewicz) [inline src/Modal/marketing-modal.mdx:16]: Lol, even though this is straight from Figma, it reads a bit weird due IMO. Maybe:

```
``MarketingModal`` is similar to the alert modal, providing friendly messaging to our users to encourage greater engagement than the simpler confirmation dialogs. They should enable platform understanding, highlight opportunities, and support workflow engagement.

The ``MarketingModal`` is a preconfigured `ModalDialog` that accepts an image and center aligns buttons.
```

**Review by adamstankiewicz — COMMENTED** [review]
Just one small comment to change the description of `MarketingModal` to read a bit smoother; otherwise, LGTM!

File: src/Modal/marketing-modal.mdx (hunk 1 above)
Comment (monteri) [reply src/Modal/marketing-modal.mdx:16]: Yeah, it is better, updating to this variant

**Review by adamstankiewicz — APPROVED** [review]

**Comment by edx-webhook** [comment]
@monteri 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/paragon&entry.867055334=https://github.com/edx/paragon/pull/892&entry.1484655318=https://github.com/monteri&entry.752974735=2021-11-19+11:11&entry.1917517419=2021-12-10+13:39&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**